src/core/server/elasticsearch/elasticsearch_service.ts

```diff
@@ -103,7 +109,7 @@ export class ElasticsearchService implements CoreService<ElasticsearchServiceSet
       dataClient$: clients$.pipe(map(clients => clients.dataClient)),
 
       createClient: (type: string, clientConfig: ElasticsearchClientConfig) => {
-        return this.createClusterClient(type, clientConfig);
+        return this.createClusterClient(type, clientConfig, deps.http.auth.getAuthHeaders);
```

Comment:
should we extend all external cluster clients with this functionality or we should make it configurable?

Reply:
The only example I can contrive where we may want to call `asScoped` on a `ClusterClient` with a `KibanaRequest` and _not_ get the auth from the internal HTTP server is the proxy plugin which has it's own HTTPServer. That said, that plugin is not planning to use header/credential forwarding, but it may make the `createNewServer` API behave strangely to 3rd party consumers.

Reply:
agree I want to extract all Kibana specific logic from HttpServer to HttpService in https://github.com/elastic/kibana/issues/38092